includes/api/class-wc-rest-products-controller.php

```diff
@@ -1422,239 +1460,146 @@ protected function save_variations_data( $product, $request, $single_variation =
-				} elseif ( isset( $request['inventory_delta'] ) ) {
-					$stock_quantity  = wc_stock_amount( get_post_meta( $variation_id, '_stock', true ) );
-					$stock_quantity += wc_stock_amount( $request['inventory_delta'] );
+			$backorders = $variation->get_backorders();
+			if ( isset( $data['backorders'] ) ) {
+				$backorders = $data['backorders'];
+			}
+			$variation->set_backorders( $backorders );
 
-					wc_update_product_stock( $variation_id, wc_stock_amount( $stock_quantity ) );
+			if ( $manage_stock ) {
+				if ( isset( $data['stock_quantity'] ) ) {
+					$variation->set_stock_quantity( $data['stock_quantity'] );
+				} elseif ( isset( $data['inventory_delta'] ) ) {
+					$stock_quantity  = wc_stock_amount( $variation->get_stock() );
+					$stock_quantity += wc_stock_amount( $data['inventory_delta'] );
+					$variation->set_stock_quantity( $stock_quantity );
 				}
 			} else {
-				delete_post_meta( $variation_id, '_backorders' );
-				wc_update_product_stock( $variation_id, '' );
+				$variation->set_backorders( 'no' );
+				$variation->set_stock_quantity( '' );
 			}
 
 			// Regular Price.
-			if ( isset( $variation['regular_price'] ) ) {
-				$regular_price = ( '' === $variation['regular_price'] ) ? '' : $variation['regular_price'];
-			} else {
-				$regular_price = get_post_meta( $variation_id, '_regular_price', true );
+			if ( isset( $data['regular_price'] ) ) {
+				$variation->set_regular_price( $data['regular_price'] );
 			}
 
 			// Sale Price.
-			if ( isset( $variation['sale_price'] ) ) {
-				$sale_price = ( '' === $variation['sale_price'] ) ? '' : $variation['sale_price'];
-			} else {
-				$sale_price = get_post_meta( $variation_id, '_sale_price', true );
+			if ( isset( $data['sale_price'] ) ) {
+				$variation->set_sale_price( $data['sale_price'] );
 			}
 
-			if ( isset( $variation['date_on_sale_from'] ) ) {
-				$date_from = $variation['date_on_sale_from'];
-			} else {
-				$date_from = get_post_meta( $variation_id, '_sale_price_dates_from', true );
-				$date_from = ( '' === $date_from ) ? '' : date( 'Y-m-d', $date_from );
+			if ( isset( $data['date_on_sale_from'] ) ) {
+				$variation->set_date_on_sale_from( $data['date_on_sale_from'] );
 			}
 
-			if ( isset( $variation['date_on_sale_to'] ) ) {
-				$date_to = $variation['date_on_sale_to'];
-			} else {
-				$date_to = get_post_meta( $variation_id, '_sale_price_dates_to', true );
-				$date_to = ( '' === $date_to ) ? '' : date( 'Y-m-d', $date_to );
+			if ( isset( $data['date_on_sale_to'] ) ) {
+				$variation->set_date_on_sale_to( $data['date_on_sale_to'] );
 			}
 
-			_wc_save_product_price( $variation_id, $regular_price, $sale_price, $date_from, $date_to );
-
 			// Tax class.
-			if ( isset( $variation['tax_class'] ) ) {
-				if ( 'parent' !== $variation['tax_class'] ) {
-					update_post_meta( $variation_id, '_tax_class', wc_clean( $variation['tax_class'] ) );
+			if ( isset( $data['tax_class'] ) ) {
+				if ( 'parent' !== $data['tax_class'] ) {
```

Comment:
This can and should be set to 'parent' if thats what is needed. CRUD takes care of inheritance. `''` is standard tax class.